diff --git a/node_modules/@cfpreview/pages-e2e-test-runner-cli/scripts/run-test.ts b/node_modules/@cfpreview/pages-e2e-test-runner-cli/scripts/run-test.ts
index 1424a14..3cec4be 100644
--- a/node_modules/@cfpreview/pages-e2e-test-runner-cli/scripts/run-test.ts
+++ b/node_modules/@cfpreview/pages-e2e-test-runner-cli/scripts/run-test.ts
@@ -119,39 +119,81 @@ This is going to be evaluated on ${environment}, using ${TRIGGER} as the trigger
 		});
 	}
 
-	const deployedFixtures = Object.fromEntries(
-		await Promise.all(
-			fixtures.map(async (fixture) => {
-				const fixtureLogger = Logger.from({
-					logger,
-					label: `[${fixture}]`,
-				});
-				const { directory, config: fixtureConfig } = await setUpFixture({
-					timestamp: startTimestamp,
-					logger: fixtureLogger,
-					fixture,
-				});
-				const { config: featuresConfig } = await setUpFeatures({
-					logger: fixtureLogger,
-					fixture,
-					features: fixtureConfig.features,
-					directory,
-				});
-				const { url } = await createDeployment({
-					timestamp: startTimestamp,
-					environment,
-					trigger,
-					logger: fixtureLogger,
-					teardownService,
-					fixture,
-					fixtureConfig,
-					featuresConfig,
-					directory,
-				});
-				return [fixture, { url, features: fixtureConfig.features }];
-			}) as Promise<[string, { url: string; features: Feature[] }]>[]
-		)
-	);
+	let deployedFixtures: {
+		[k: string]: {
+			url: string;
+			features: Feature[];
+		};
+	} = {};
+	if(environment !== Environment.Local) {
+		deployedFixtures = Object.fromEntries(
+			await Promise.all(
+				fixtures.map(async (fixture, idx) => {
+					const fixtureLogger = Logger.from({
+						logger,
+						label: `[${fixture}]`,
+					});
+					const { directory, config: fixtureConfig } = await setUpFixture({
+						timestamp: startTimestamp,
+						logger: fixtureLogger,
+						fixture,
+					});
+					const { config: featuresConfig } = await setUpFeatures({
+						logger: fixtureLogger,
+						fixture,
+						features: fixtureConfig.features,
+						directory,
+					});
+
+					const { url } = await createDeployment({
+						timestamp: startTimestamp,
+						environment,
+						trigger,
+						logger: fixtureLogger,
+						teardownService,
+						fixture,
+						fixtureConfig,
+						featuresConfig,
+						directory,
+					});
+					return [fixture, { url, features: fixtureConfig.features }];
+				}) as Promise<[string, { url: string; features: Feature[] }]>[]
+			)
+		);
+	} else {
+		console.log('⚠️⚠️⚠️ Detected local environment so all deployments are going to be performed sequentially ⚠️⚠️⚠️');
+
+		for(const fixture of fixtures) {
+			const fixtureLogger = Logger.from({
+				logger,
+				label: `[${fixture}]`,
+			});
+			const { directory, config: fixtureConfig } = await setUpFixture({
+				timestamp: startTimestamp,
+				logger: fixtureLogger,
+				fixture,
+			});
+			const { config: featuresConfig } = await setUpFeatures({
+				logger: fixtureLogger,
+				fixture,
+				features: fixtureConfig.features,
+				directory,
+			});
+
+			const { url } = await createDeployment({
+				timestamp: startTimestamp,
+				environment,
+				trigger,
+				logger: fixtureLogger,
+				teardownService,
+				fixture,
+				fixtureConfig,
+				featuresConfig,
+				directory,
+			});
+			deployedFixtures[fixture] = { url, features: fixtureConfig.features };
+		}
+	}
 
 	const testResults = await runTests({
 		logger,
